### PR TITLE
integrating apple session policy batch deletes hotfix tests into cephci

### PR DESCRIPTION
**Pull request by hmaheswa**
this PR adds support for testing policy actions for features like notification, encryption, tagging, lifecycle, bucket website other than batch deletes.
added support for testing bucket policy, sts assume-role with session policy, aswi role policy (both allow and deny effect).

depends on: https://github.com/red-hat-storage/ceph-qe-scripts/pull/615

hotfix bz: https://bugzilla.redhat.com/show_bug.cgi?id=2284153

polarion id: https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83593390

found below bz through automation:
https://bugzilla.redhat.com/show_bug.cgi?id=2302541

automated below bz verification as well, as they are targetted to 8.1 commented the test in lower versions
https://bugzilla.redhat.com/show_bug.cgi?id=2302541
https://bugzilla.redhat.com/show_bug.cgi?id=2293233

pass logs: http://magna002.ceph.redhat.com/cephci-jenkins/hsm/PR_apple_sts_batch_deletes/

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Review by psathyan — COMMENTED** [review]
nit 1: `sanity_rgw.py` does not honor timeout, hence I would recommend not setting this value. Alternatively, you could fix the test module.

nit 2:  when using `git log` or cli ... tracability is tard difficult hence would suggest having meaningful commits. That is we can use multiple commits in this PR to segragate the changes.

File: suites/pacific/rgw/tier-1-extn_rgw.yaml [hunk 1]
```diff
@@ -138,6 +138,24 @@ tests:
...
         timeout: 500
+  - test:
+      name: STS test to verify session policy allow actions
+      desc: STS test to verify session policy allow actions
+      polarion-id: CEPH-83593390
+      module: sanity_rgw.py
+      config:
+        script-name: test_sts_using_boto_session_policy.py
+        config-file-name: test_sts_using_boto_verify_session_policy_allow_actions.yaml
+        timeout: 500
```
[the remark this answers is not in the document]
Comment (hmaheswa) [reply suites/pacific/rgw/tier-1-extn_rgw.yaml]: @psathyan , removed the timeout

File: suites/pacific/rgw/tier-1-extn_rgw.yaml [hunk 2]
```diff
@@ -138,6 +138,22 @@ tests:
         script-name: test_sts_using_boto_session_policy.py
         config-file-name: test_sts_using_boto_restricted_session_policy.yaml
         timeout: 500
+  - test:
```
Comment (anrao19) [inline suites/pacific/rgw/tier-1-extn_rgw.yaml:141]: Timeout has been removed from quincy, reef and squide but not from pacific.
if there any changes in pacific suite, please remove timeout from all testcase of that suite.
When not sure on the timeout needed

Comment (hmaheswa) [reply suites/pacific/rgw/tier-1-extn_rgw.yaml:141]: @anrao19 
I removed timeout from only my tests in the PR
there are many such places where timeout is present in all of our RGW suites
we can take it in another PR know
why stressing in this PR

Comment (hmaheswa) [reply suites/pacific/rgw/tier-1-extn_rgw.yaml:137]: done, pls review

**Review by psathyan — COMMENTED** [review]
/lgtm

**Review by anrao19 — APPROVED** [review]
LGTM

**Review by anrao19 — APPROVED** [review]
LGTM

**Review by viduship — APPROVED** [review]

**Review by ckulal — APPROVED** [review]

**Pull request merged**